### PR TITLE
Allow Unicode strings as item symbols. 

### PR DESCRIPTION
**Pull request by BevapDin**
Basically like the monster symbols. Any UTF-8 string should work, as long as it is exactly 1 console cell width.

### CODE REVIEW

File: src/item_factory.cpp [hunk 1]
```diff
@@ -545,8 +546,10 @@ void Item_factory::check_definitions() const
             }
         }
 
-        if( type->sym == 0 ) {
+        if( type->sym.empty() ) {
             msg << "symbol not defined" << "\n";
+        } else if( utf8_width( type->sym ) != 1 ) {
+            msg << "symbol must be exactly one console cell width" << "\n";
```
Comment (codemime) [inline src/item_factory.cpp]: IMHO a bit vague. I'd write something like `"only one symbol allowed"`

Comment (BevapDin) [inline src/item_factory.cpp]: Well, your suggestion would state something incorrect. One can have as many Unicode code points as one likes. Only together they must consume one character on the console. For example combination marks have a width of 0 and can be combined with other code points without increasing displayed length.

Comment (codemime) [inline src/item_factory.cpp]: I guess you're right. Sorry for the irrelevant remark.

**Comment by mugling** [comment]
Results in display distortion. Can you include a suitable test item?

**Pull request merged**